src/sys/socket/sockopt.rs

```diff
@@ -1745,7 +1745,13 @@ impl<T: AsMut<[u8]>> Get<OsString> for GetOsString<T> {
     unsafe fn assume_init(self) -> OsString {
         let len = self.len as usize;
         let mut v = unsafe { self.val.assume_init() };
-        OsStr::from_bytes(&v.as_mut()[0..len]).to_owned()
+        if let Ok(cs) = CStr::from_bytes_until_nul(&v.as_mut()[0..len]) {
+            OsStr::from_bytes(cs.to_bytes())
+        } else {
+            // The OS returned a non-NUL-terminated string
+            OsStr::from_bytes(&v.as_mut()[0..len])
+        }
```

Comment:
Considering that this behavior is version-dependent, i.e., some contributors, with specific kernel versions, may find this check useless, perhaps we should document why we do this here 🤔